aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1317,6 +1317,28 @@ std::vector<Tensor> unsafe_split(const Tensor& self, int64_t split_size, int64_t
   return result;
 }
 
+std::vector<Tensor> hsplit(const Tensor& self, int64_t split_size, c10::optional<int64_t> dummy) {
+  TORCH_CHECK(!dummy.has_value(), "torch.hsplit expects dummy variable never take a value.")
```

Comment:
I realize I was the one who used the word "dummy", but let's call this value "sentinel"

Reply:
From a user-facing perspective the variable "sentinel" doesn't exist, so this error message needs to be updated to reflect the user's intent and educate them about the correct usage of hsplit().

"torch.hsplit takes either a single index or a list of indices, but got two indices outside of a list!"